proof/crefine/AARCH64/Syscall_C.thy

```diff
@@ -1785,20 +1786,32 @@ proof -
     apply (subst unat_add_lem' ; clarsimp simp: unat_of_nat_ctz_mw)
     using word_ctz_le[where w=w, simplified] by (auto simp: unat_of_nat_eq)
 
-  have eisr_calc_le:
-    "eisr0 = 0 \<longrightarrow> eisr1 \<noteq> 0
-     \<Longrightarrow> eisr_calc eisr0 eisr1 \<le> 63"
-    for eisr0 and eisr1
-    using word_ctz_le[where w=eisr0]  word_ctz_less[where w=eisr1]
-    by (clarsimp simp: eisr_calc_def split: if_splits)
-
   have of_nat_word_ctz_0x21helper:
     "0x21 + word_of_nat (word_ctz w) \<noteq> (0 :: int_word)" for w :: "32 word"
     apply (subst unat_arith_simps, simp)
     apply (subst unat_add_lem'; clarsimp simp: unat_of_nat_ctz_smw)
     using word_ctz_le[where w=w, simplified]
     by simp
 
+  have maxListRegs_unat_id:
+    "\<And>n. n < max_armKSGICVCPUNumListRegs \<Longrightarrow> unat (of_nat n :: machine_word) = n"
+    by (simp add: max_armKSGICVCPUNumListRegs_val unat_of_nat)
+
+  have maxListRegs_sint_ge_0:
+    "\<And>n. n < max_armKSGICVCPUNumListRegs \<Longrightarrow> 0 \<le> sint (word_of_nat n :: int_word)"
+    by (simp add: max_armKSGICVCPUNumListRegs_val int_eq_sint)
+
+  have maxListRegs_sint_less:
+    "\<And>n. n < max_armKSGICVCPUNumListRegs \<Longrightarrow>
+          sint (word_of_nat n :: int_word) < int max_armKSGICVCPUNumListRegs"
+    by (simp add: max_armKSGICVCPUNumListRegs_val int_eq_sint)
+
+  (* needs "simp del: of_nat_add", because of rhs *)
```

Comment:
This should maybe say `(* needs "simp del: of_nat_add" when used, because of rhs *)`. I was confused when I first read it and there wasn't a `simp del` immediately visible.